js/src/edit-block/components/field-panel.js

```diff
@@ -112,14 +112,15 @@ const FieldPanel = ( {
 								if ( isNewField ) {
 									didAutoSlug.current = true;
 									const newName = convertToSlug( event.target.value );
-									changedField.name = newName;
+									const newDeDuplicatedName = changeFieldSettings( fieldToChange, { ...changedField, name: newName } );
+
 									setSelectedField( {
 										...fieldToChange,
-										name: newName,
+										name: newDeDuplicatedName,
```

Comment:
When changing a field name, this changed the `selectedField` to the new name.

The `selectedField` is what shows in the side panel:

<img width="651" alt="selected-field" src="https://user-images.githubusercontent.com/4063887/105551376-47a2d380-5cc8-11eb-86e8-926ffd1006d2.png">

Reply:
This now uses a deduplicated name, in case there's another field of the same name.